### PR TITLE
[WIP] JS sighash implementation

### PR DESCRIPTION
**Pull request by prestwich**
implement sighash in JS for later use in other projects.

- [x] test vectors
- [x] docstrings

some tests:
https://github.com/summa-tx/riemann-rs/blob/master/bitcoin/src/types/transactions.rs#L760-L794
https://github.com/summa-tx/riemann-rs/blob/master/bitcoin/src/types/transactions.rs#L796-L830

### CODE REVIEW

**Comment by prestwich** [comment]
some docstrings:

https://github.com/summa-tx/riemann-rs/blob/master/bitcoin/src/types/transactions.rs#L453-L515
https://github.com/summa-tx/riemann-rs/blob/master/bitcoin/src/types/transactions.rs#L392-L425

File: js/src/sighash.js [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+export function parseVin(vin) {
+  const { number: nIns } = BTCUtils.parseVarInt(vin);
+  const inputs = [];
+  for (let i = 0; i < nIns; i += 1) {
+    inputs.push(BTCUtils.extractInputAtIndex(vin, i));
+  }
+  return inputs;
+}
+
+// returns an array of uint8arrays.
```
Comment (ErinHales) [inline js/src/sighash.js:71]: Oops, delete this line

File: js/src/sighash.js [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+
+  const sequences = inputs.map(BTCUtils.extractSequenceLegacy);
+  for (let i = 0; i < sequences.length; i += 1) {
+    const disableFlag = (sequences[i] & BigInt(0x80)) === BigInt(0x80);
+    if (!disableFlag) return true;
+  }
+  return false;
+}
+
+// TODO: make typedef for tx
```
Comment (ErinHales) [inline js/src/sighash.js:191]: Delete this line

File: js/src/sighash.js [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+/**
+ *
+ * Hashes prevouts according to BIP143 semantics.
+ * 
+ * For BIP143 (Witness and Compatibility sighash) documentation, see here:
+ * - https://github.com/bitcoin/bips/blob/master/bip-0143.mediawiki
+ *
+ * @param {array}         inputs An array of inputs (type Uint8Array)
+ * @param {number}        flag The sighash flag
+ * @returns {Uint8Array}  The hash of the Prevouts
```
Comment (ErinHales) [inline js/src/sighash.js:97]: Should this be `BIP143 hashPrevouts` like the others?

**Comment by ErinHales** [comment]
Minor things on my part.  The rest looked good!

**Pull request merged**